v2/tools/generator/internal/config/type_configuration.go

```diff
@@ -54,11 +55,18 @@ const (
 	isResourceTag               = "$isResource"               // Boolean specifying whether a particular type is a resource or not.
 	manualConfigsTag            = "$manualConfigs"            // A set of strings specifying which config map fields should be generated (to be filled out by resource extension)
 	nameInNextVersionTag        = "$nameInNextVersion"        // String specifying a type or property name change in the next version
+	operatorSpecPropertiesTag   = "$operatorSpecProperties"   // A set of additional properties to inject into the operatorSpec of a resource
 	renameTo                    = "$renameTo"                 // String specifying the new name of a type
 	resourceEmbeddedInParentTag = "$resourceEmbeddedInParent" // String specifying resource name of parent
 	supportedFromTag            = "$supportedFrom"            // Label specifying the first ASO release supporting the resource
 )
 
+type OperatorSpecPropertyConfiguration struct {
+	Name        string `yaml:"name"`        // Name of the new property
```

Comment:
Not that it's necessarily going to come up because we don't serialize this, but consider `omitempty` on the bits anyway, just as a general good practice?

Reply:
Done.